core/src/mongosqltranslate_data_types.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{col_metadata::ResultSetSchema, Error, Result};
+use bson::{doc, Bson, Document};
+use serde::ser::SerializeMap;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeSet;
+
+pub trait CommandName {
+    fn command_name() -> &'static str;
+}
+
+#[derive(Debug)]
+pub struct Command<T> {
+    pub options: T,
+}
+impl<T> serde::ser::Serialize for Command<T>
+where
+    T: CommandName + serde::ser::Serialize,
+{
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("command", T::command_name())?;
+        map.serialize_entry("options", &self.options)?;
+        map.end()
+    }
+}
+
+impl<T> Command<T> {
+    pub fn new(options: T) -> Self {
+        Self { options }
+    }
+}
+
```

Comment:
If we add a `From` impl, the usage becomes a lot nicer. Just a suggestion, it's not a blocker. My other suggestions will be built on this one.

```suggestion
impl<T: CommandName> From<T> for Command<T> {
    fn from(options: T) -> Self {
        Self { options }
    }
}
```